test/device/common.hpp

```diff
@@ -30,7 +30,18 @@
 template <typename T>
 __device__ inline double toDouble(T const& val)
 {
-    return static_cast<double>(static_cast<float>(val));
+    using hiptensor::float32_t;
+    using hiptensor::float64_t;
+
+    // Small floating point types need to convert up to single before double.
+    if constexpr((sizeof(T) < sizeof(float32_t)) && std::is_floating_point_v<T>)
```

Comment:
I noticed that hip does not define is_floating_point_v for bf16

`std::is_floating_point_v<bf16>  == false`
`std::is_floating_point_v<_Float16 >  == true`

Reply:
Fixed